From 45599fd8fe420ede67b0a28767a335d1d203f9cf Mon Sep 17 00:00:00 2001
From: Dag Bakke <dag@bakke.com>
Date: Mon, 30 May 2022 11:57:06 +0200
Subject: [PATCH] overlays: dht11: make sensor readable via hwmon

Add a patch to make the sensor readable via the IIO/hwmon bridge.

Signed-off-by: Dag Bakke <dag@bakke.com>
---
 arch/arm/boot/dts/overlays/dht11-overlay.dts | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boot/dts/overlays/dht11-overlay.dts b/arch/arm/boot/dts/overlays/dht11-overlay.dts
index 6feeeb402493..8b0fc6b7a3cb 100644
--- a/arch/arm/boot/dts/overlays/dht11-overlay.dts
+++ b/arch/arm/boot/dts/overlays/dht11-overlay.dts
@@ -10,13 +10,19 @@ / {
 	fragment@0 {
 		target-path = "/";
 		__overlay__ {
-
-			dht11: dht11@0 {
+			dht11: dht11@4 {
 				compatible = "dht11";
 				pinctrl-names = "default";
 				pinctrl-0 = <&dht11_pins>;
 				gpios = <&gpio 4 0>;
 				status = "okay";
+				#io-channel-cells = <1>;
+			};
+
+			iio: iio-hwmon@4 {
+				compatible = "iio-hwmon";
+				status = "okay";
+				io-channels = <&dht11 0>, <&dht11 1>;
 			};
 		};
 	};
@@ -24,7 +30,7 @@ dht11: dht11@0 {
 	fragment@1 {
 		target = <&gpio>;
 		__overlay__ {
-			dht11_pins: dht11_pins@0 {
+			dht11_pins: dht11_pins@4 {
 				brcm,pins = <4>;
 				brcm,function = <0>; // in
 				brcm,pull = <0>; // off
@@ -36,6 +42,7 @@ __overrides__ {
 		gpiopin = <&dht11_pins>,"brcm,pins:0",
 			<&dht11_pins>, "reg:0",
 			<&dht11>,"gpios:4",
-			<&dht11>,"reg:0";
+			<&dht11>,"reg:0",
+			<&iio>,"reg:0";
 	};
 };
-- 
2.32.0

